cdc/owner/owner.go

```diff
@@ -0,0 +1,271 @@
+
+// All AdminJob types
+const (
+	ownerJobTypeRebalance ownerJobType = iota
+	ownerJobTypeManualSchedule
+	ownerJobTypeAdminJob
+	ownerJobTypeDebugInfo
+)
+
+type ownerJob struct {
+	tp           ownerJobType
+	changefeedID model.ChangeFeedID
+
+	// for ManualSchedule only
+	targetCaptureID model.CaptureID
+	// for ManualSchedule only
+	tableID model.TableID
+
+	// for Admin Job only
+	adminJob *model.AdminJob
+
+	// for debug info only
+	httpWriter http.ResponseWriter
+
+	done chan struct{}
+}
+
+// Owner manages many changefeeds
+// All published functions is THREAD-SAFE, except for Tick, Tick is only used for etcd worker
+type Owner struct {
+	changefeeds map[model.ChangeFeedID]*changefeed
+
+	gcManager *gcManager
+
+	ownerJobQueue   []*ownerJob
+	ownerJobQueueMu sync.Mutex
+	lastTickTime    time.Time
+
+	close int32
+
+	newChangefeed func(id model.ChangeFeedID, gcManager *gcManager) *changefeed
+}
+
+// NewOwner creates a new Owner
+func NewOwner() *Owner {
+	return &Owner{
+		changefeeds:   make(map[model.ChangeFeedID]*changefeed),
+		gcManager:     newGCManager(),
+		lastTickTime:  time.Now(),
+		newChangefeed: newChangefeed,
+	}
+}
+
+// NewOwner4Test creates a new Owner for test
+func NewOwner4Test(
+	newDDLPuller func(ctx cdcContext.Context, startTs uint64) (DDLPuller, error),
+	newSink func(ctx cdcContext.Context) (AsyncSink, error)) *Owner {
+	o := NewOwner()
+	o.newChangefeed = func(id model.ChangeFeedID, gcManager *gcManager) *changefeed {
+		return newChangefeed4Test(id, gcManager, newDDLPuller, newSink)
+	}
+	return o
+}
+
+// Tick implements the Reactor interface
+func (o *Owner) Tick(stdCtx context.Context, rawState orchestrator.ReactorState) (nextState orchestrator.ReactorState, err error) {
+	failpoint.Inject("owner-run-with-error", func() {
+		failpoint.Return(nil, errors.New("owner run with injected error"))
+	})
+	failpoint.Inject("sleep-in-owner-tick", nil)
+	ctx := stdCtx.(cdcContext.Context)
+	state := rawState.(*model.GlobalReactorState)
+	o.updateMetrics(state)
+	state.CheckCaptureAlive(ctx.GlobalVars().CaptureInfo.ID)
+	err = o.gcManager.updateGCSafePoint(ctx, state)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	o.handleJob()
+	for changefeedID, changefeedState := range state.Changefeeds {
+		if changefeedState.Info == nil {
+			o.cleanUpChangefeed(changefeedState)
+			continue
+		}
+		ctx = cdcContext.WithChangefeedVars(ctx, &cdcContext.ChangefeedVars{
+			ID:   changefeedID,
+			Info: changefeedState.Info,
+		})
+		cfReactor, exist := o.changefeeds[changefeedID]
+		if !exist {
+			cfReactor = o.newChangefeed(changefeedID, o.gcManager)
+			o.changefeeds[changefeedID] = cfReactor
+		}
+		cfReactor.Tick(ctx, changefeedState, state.Captures)
+	}
+	if len(o.changefeeds) != len(state.Changefeeds) {
+		for changefeedID, cfReactor := range o.changefeeds {
+			if _, exist := state.Changefeeds[changefeedID]; exist {
+				continue
+			}
+			cfReactor.Close()
+			delete(o.changefeeds, changefeedID)
```

Comment:
the `cfReactor` is not used afterward, I notice there exist some operations in `func (c *changefeed) releaseResources()`, such as
```
c.metricsChangefeedCheckpointTsGauge = nil
c.metricsChangefeedCheckpointTsLagGauge = nil
c.initialized = false  
```
are these useless